src/shared/utils/LoggingUtils/generateErrorBasedOnEnvironment.ts

```diff
@@ -0,0 +1,33 @@
+import { getNow } from '../datetime/now';
+
+export default function generateErrorBasedOnCurrentEnvironment(
+  status: number,
+  message: string,
+  stack,
+  url,
+  environment: string,
+) {
+  let expectedErrorToBeReturned = {};
+
+  if (environment === 'development') {
+    expectedErrorToBeReturned = {
+      statusCode: status,
+      message: message,
+      stack,
+      timestamp: getNow(),
+      path: url,
+    };
+  } else {
+    if (status >= 500) {
+      message = 'Internal Server Error';
+    }
+
```

Comment:
Why is that necessary ?
Also, if it is necessary, it doesn't change the status code.

Reply:
Because I shouldn't reveal any information about the error when its code is >= 500.
It changes the message that will be send to the user, and its effect will be shown when its merged.
Or do you mean something else?

Reply:
Ok but then you will need to change the status code too

Reply:
IMO, I don't think sending the real status code to the user back in the production environment.
The error -if occurred-, we should return the least possible information to the user, yet descriptive that an error occurred without showing any details.
What do you think?